crypto/secp256r1/verifier.go

```diff
@@ -2,21 +2,36 @@ package secp256r1
 
 import (
 	"crypto/ecdsa"
+	"crypto/elliptic"
+	"errors"
 	"math/big"
+
+	"github.com/ethereum/go-ethereum/common"
 )
 
+var secp256k1halfN = new(big.Int).Div(elliptic.P256().Params().N, common.Big2)
+
+// Check the malleability issue, return false if malleable
+func checkMalleability(s *big.Int) bool {
```

Comment:
let's ask those question in upstream repo.  I think we will need more input from the open source community especially from cryptographers.

Reply:
Using the lower-S form of ECDSA signatures is a common solution in Geth. However, you're correct—we should make sure we're compatible with upstream repo.